awx/main/utils/plugins.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Ansible, Inc.
+# All Rights Reserved.
+
+"""
+This module contains the code responsible for extracting the lists of dynamically discovered plugins.
+"""
+
+from functools import cache
+
+
+@cache
+def discover_available_cloud_provider_plugin_names() -> list[str]:
+    """Return a list of cloud plugin names available in runtime.
+
+    The discovery result is cached since it does not change throughout
+    the life cycle of the server run.
+
+    :returns: List of plugin cloud names.
+    :rtype: list[str]
+    """
+    from awx.main.models.inventory import InventorySourceOptions
+
+    return list(InventorySourceOptions.injectors.keys())
+
+
+def compute_cloud_inventory_sources() -> dict[str, str]:
+    """Return a dictionary of cloud provider plugin names
+    available plus source control management.
+
+    :returns: Dictionary of plugin cloud names plus source control.
+    :rtype: dict[str, str]
+    """
+    # Get the list of plugin names
+    plugins = discover_available_cloud_provider_plugin_names()
+
+    # Create a dictionary of plugin names
+    cloud_sources = {plugin: plugin for plugin in plugins}
+    cloud_sources['scm'] = 'Source Control Management'
+
+    return cloud_sources  # Return the complete dictionary
```

Comment:
Let's compose the dict in one go:
```suggestion
    return dict(zip(plugins, plugins), scm='Source Control Management')
```